checks/raw/pinned_dependencies.go

```diff
@@ -187,10 +189,20 @@ func isDockerfile(pathfn string, content []byte) bool {
 }
 
 func collectDockerfilePinning(c *checker.CheckRequest, r *checker.PinningDependenciesData) error {
-	return fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
+	err := fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
 		Pattern:       "*Dockerfile*",
 		CaseSensitive: false,
 	}, validateDockerfilesPinning, r)
+	if err != nil {
+		return err
+	}
+	//nolint:errcheck
+	remediationMetadata, _ := remediation.New(c)
+	for i := range r.Dependencies {
+		rr := r.Dependencies[i]
+		rr.Remediation = generateRemediation(remediationMetadata, &rr)
```

Comment:
i'm curious if this needs to be
```
rr := &r.Dependencies[i]
```

in order to modify the original, instead of a copy

Reply:
Thanks for catching that. Done in https://github.com/ossf/scorecard/pull/3632/commits/227bcc14c019076e3d94a67f30b5e9fada781b7c